brave/brave-core
# P2A for Brave Ads POC

Resolves https://github.com/brave/brave-browser/issues/10081

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:
use flags to decrease upload and rotation time `--p3a-upload-interval-seconds=5 --p3a-rotation-interval-seconds=160`

case 1: send p2a message

- fresh profile, open browser with vpn to get us catalog, enable ads
- navigate to brave.com to have a page classification for ads
- let browser enter idle state, move mouse and trigger ad > observe p2a log
- wait a couple of minutes for p3a/p2a uploader to trigger > observe p2a message in charles proxy

case 2: prevent sending p2a message when disabling ads

- fresh profile, open browser with vpn to get us catalog, enable ads
- navigate to brave.com to have a page classification for ads
- let browser enter idle state, move mouse and trigger ad > observe p2a log
- disable ads and/or rewards > observe another p2a log
- wait a couple of minutes for p3a/p2a uploader to trigger > observe lack of p2a message in charles proxy

case 3:

- re-enable ads and observe new p2a logs in local state and/or charles

case 4:

- instead of re-enabling, create a new profile and observe p2a logs/uploads

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
